eb_hooks.py

```diff
@@ -353,6 +384,86 @@ def parse_hook_CP2K_remove_deps_for_aarch64(ec, *args, **kwargs):
+        ec['modluafooter'] = 'if (not os.getenv("%s")) then LmodError("%s") end' % (env_varname, errmsg)
+
+
+def pre_fetch_hook(self, *args, **kwargs):
+    """Main pre fetch hook: trigger custom functions based on software name."""
+    if self.name in PRE_FETCH_HOOKS:
+        PRE_FETCH_HOOKS[ec.name](self, *args, **kwargs)
+
+    # Always trigger this one, regardless of self.name
+    cpu_target = get_eessi_envvar('EESSI_SOFTWARE_SUBDIR')
+    if cpu_target == CPU_TARGET_ZEN4:
+        pre_fetch_hook_zen4_gcccore1220(self, *args, **kwargs)
+
+
+def pre_fetch_hook_zen4_gcccore1220(self, *args, **kwargs):
+    """Use --force --module-only if building a foss-2022b-based EasyConfig for Zen4.
+    This toolchain will not be supported on Zen4, so we will generate a modulefile
+    and have it print an LmodError.
+    """
+    if is_gcccore_1220_based(self.name, self.version, self.toolchain.name, self.toolchain.version):
+        if hasattr(self, EESSI_MODULE_ONLY_ATTR):
+            raise EasyBuildError("'self' already has attribute %s! Can't use pre_fetch hook.",
+                                 EESSI_MODULE_ONLY_ATTR)
+        setattr(self, EESSI_MODULE_ONLY_ATTR, build_option('module_only'))
+        update_build_option('module_only', 'True')
+        print_msg("Updated build option 'module-only' to 'True'")
+
+        if hasattr(self, EESSI_FORCE_ATTR):
+            raise EasyBuildError("'self' already has attribute %s! Can't use pre_fetch hook.",
+                                 EESSI_FORCE_ATTR)
+        setattr(self, EESSI_FORCE_ATTR, build_option('force'))
+        update_build_option('force', 'True')
+        print_msg("Updated build option 'force' to 'True'")
+
+
+def post_module_hook_zen4_gcccore1220(self, *args, **kwargs):
+    """Revert changes from pre_fetch_hook_zen4_gcccore1220"""
+    if is_gcccore_1220_based(self.name, self.version, self.toolchain.name, self.toolchain.version):
+        if hasattr(self, EESSI_MODULE_ONLY_ATTR):
+            update_build_option('module_only', getattr(self, EESSI_MODULE_ONLY_ATTR))
+            print_msg("Restored original build option 'module_only' to %s" % getattr(self, EESSI_MODULE_ONLY_ATTR))
+        else:
+            raise EasyBuildError("Cannot restore module_only to it's original value: 'self' is missing attribute %s.",
+                                 EESSI_MODULE_ONLY_ATTR)
+
+        if hasattr(self, EESSI_FORCE_ATTR):
+            update_build_option('force', getattr(self, EESSI_FORCE_ATTR))
+            print_msg("Restored original build option 'force' to %s" % getattr(self, EESSI_FORCE_ATTR))
+        else:
+            raise EasyBuildError("Cannot restore force to it's original value: 'self' is misisng attribute %s.",
+                                 EESSI_FORCE_ATTR)
+
+
+# We do this as early as possible - and remove it all the way in the last step hook (post_testcases_hook)
+def pre_prepare_hook_ignore_zen4_gcccore1220_error(self, *args, **kwargs):
```

Comment:
Is there really any harm in collapsing this into your pre_fetch hook? It would be nice to keep the setting and unsetting unified for our future selves to understand better what would need to be changed.

Reply:
Or is the environment variable not persistent?